src/main/java/de/dennisguse/opentracks/TrackDeleteActivity.java

```diff
@@ -44,53 +42,52 @@ public void onCreate(Bundle savedInstanceState) {
         setResult(RESULT_CANCELED);
 
         Intent intent = getIntent();
-        long[] trackIds = intent.getLongArrayExtra(EXTRA_TRACK_IDS);
-
-        Object retained = getLastNonConfigurationInstance();
-        if (retained instanceof DeleteAsyncTask) {
-            deleteAsyncTask = (DeleteAsyncTask) retained;
-            deleteAsyncTask.setActivity(this);
-        } else {
-            deleteAsyncTask = new DeleteAsyncTask(this, trackIds);
-            deleteAsyncTask.execute();
-        }
+        trackIds = intent.getLongArrayExtra(EXTRA_TRACK_IDS);
+
+        deleteAsyncTask = new DeleteAsyncTask();
     }
 
     @Override
-    public Object onRetainNonConfigurationInstance() {
-        deleteAsyncTask.setActivity(null);
-        return deleteAsyncTask;
+    protected void onStart() {
+        super.onStart();
+        deleteAsyncTask.execute();
     }
 
     @Override
-    protected Dialog onCreateDialog(int id) {
-        if (id != DIALOG_PROGRESS_ID) {
-            return null;
-        }
-        return DialogUtils.createSpinnerProgressDialog(
-                this, R.string.track_delete_progress_message, new DialogInterface.OnCancelListener() {
-                    @Override
-                    public void onCancel(DialogInterface dialog) {
-                        deleteAsyncTask.cancel(true);
-                        dialog.dismiss();
-                        finish();
-                    }
-                });
+    protected void onStop() {
+        super.onStop();
+        deleteAsyncTask.cancel(true);
+    }
+
+    @Override
+    protected int getLayoutResId() {
+        return R.layout.track_delete;
     }
 
-    /**
-     * Invokes when the associated AsyncTask completes.
-     */
     public void onAsyncTaskCompleted() {
-        removeDialog(DIALOG_PROGRESS_ID);
         setResult(RESULT_OK);
         finish();
```

Comment:
When deleting is finished the user can see the spinner going down. Why don't hide it before `finish()`?

`findViewById(R.id.progress_bar).setVisibility(View.INVISIBLE);`